### PR TITLE
Rename Snapshotter -> Fingerprinter

### PR DESCRIPTION
**Pull request by wolfs**
Also move fingerprint related things to `org.gradle.internal.fingerprint`.

### CODE REVIEW

File: subprojects/core/src/main/java/org/gradle/internal/fingerprint/impl/SnapshotMapSerializer.java [hunk 1]
```diff
@@ -14,14 +14,15 @@
  * limitations under the License.
  */
 
-package org.gradle.api.internal.changedetection.state;
+package org.gradle.internal.fingerprint.impl;
```
Comment (lptr) [inline subprojects/core/src/main/java/org/gradle/internal/fingerprint/impl/SnapshotMapSerializer.java]: If we start using the `impl` pattern to hide implementation details for internal types, then I think we should make a proposal about it on `engineering@`.

File: subprojects/core/src/main/java/org/gradle/internal/service/scopes/TaskExecutionServices.java [hunk 2]
```diff
@@ -161,28 +161,28 @@ TaskHistoryStore createCacheAccess(Gradle gradle, CacheRepository cacheRepositor
...
 
     TaskHistoryRepository createTaskHistoryRepository(
         TaskHistoryStore cacheAccess,
-        FileCollectionSnapshotterRegistry fileCollectionSnapshotterRegistry,
+        FileCollectionFingerprinterRegistry fileCollectionFingerprinterRegistry,
         StringInterner stringInterner,
         ClassLoaderHierarchyHasher classLoaderHierarchyHasher,
         ValueSnapshotter valueSnapshotter,
-        FileCollectionSnapshotterRegistry snapshotterRegistry) {
+        FileCollectionFingerprinterRegistry fingerprinterRegistry) {
```
Comment (lptr) [inline subprojects/core/src/main/java/org/gradle/internal/service/scopes/TaskExecutionServices.java]: We are injecting the same thing twice here.

File: subprojects/core/src/main/java/org/gradle/internal/fingerprint/impl/SnapshotMapSerializer.java (hunk 1 above)
Comment (wolfs) [reply subprojects/core/src/main/java/org/gradle/internal/fingerprint/impl/SnapshotMapSerializer.java]: Sounds like a good idea.

What is your suggestion for this PR? Move everything to `org.gradle.internal.fingerprint`?

File: subprojects/core/src/main/java/org/gradle/internal/service/scopes/TaskExecutionServices.java (hunk 2 above)
Comment (wolfs) [reply subprojects/core/src/main/java/org/gradle/internal/service/scopes/TaskExecutionServices.java]: Well spotted! Fixed.

File: subprojects/core/src/main/java/org/gradle/internal/fingerprint/impl/SnapshotMapSerializer.java (hunk 1 above)
Comment (lptr) [reply subprojects/core/src/main/java/org/gradle/internal/fingerprint/impl/SnapshotMapSerializer.java]: I think it's fine to make the move to `impl` where it makes sense. We should just not forget to mention this to others. If there's disagreement, we can easily clear things up after the fact, too.

**Pull request merged**